pennylane/ops/op_modifiers.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines an operation modifier that indicates the adjoint of an operator.
+"""
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+class Modifier(Operation):
```

Comment:
Should this be inheriting from `Operator`?

Reply:
I initially tried to inherit from `Operator`, but then it wouldn't queue in the tape properly.  

Line 194 in [`QuantumTape._process_queue` ](https://github.com/PennyLaneAI/pennylane/blob/1fdbcb56e0695668f4e2987550e5999e06fbf640/pennylane/tape/tape.py#L194)  checks if the object in an `Operation`.  So if it merely inherits from `Operator` and not `Operation`, it will not be processed into the tape.

Maybe we should change the line in `_process_queue` instead?

Reply:
Oops - we should, that sounds like a bug!

Reply:
This could be a deeper pit if I remember correctly: Operations have a simple queue logic but Observables (i.e. another form of Operator) are more complicaed. When we create a Hamiltonian like `qml.Hamiltonian([0.1], PauliX(0))`, then the PauliX gets queued as well, and a system of defining one's parent observable determines if it is "really" in the queue. I think this was a design choice so we never have to mutate the queue. 

I am not sure if this is relevant, but the queueing system is very subtle. If the change fails tests, and the problem is a wrong output, this could be an indicator that the fix is not safe...

Reply:
Then again we shouldn't be queueing plain observables without owner meta-data anymore. We should always get a `MeasurementProcess` that owns an observable. If the observable has an owner, then it won't get queued into `_ops`.

Reply:
Agree. I think the only reason observables get queued is the edge case of the Pauli/Hadamard classes, which are both Operations _and_ observables.